zerver/tests/test_send_email.py

```diff
@@ -104,6 +113,64 @@ def test_initialize_connection(self) -> None:
         # 3 more calls to open as we try 3 times before giving up
         self.assertEqual(backend.open.call_count, 6)
 
+    def test_max_connection_lifetime(self) -> None:
+        backend = mock.MagicMock(spec=CustomSMTPEmailBackend)
+        backend.connection = mock.MagicMock(spec=SMTP)
+
+        backend.open.return_value = False
+        backend.connection.noop.return_value = [250]
+
+        # Old connection is open, but we still reopen a new connection because max connection lifetime is 0.
+        with mock.patch.object(backend, "opened_at", timezone_now(), create=True), self.settings(
+            EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES=0
+        ):
+            initialize_connection(backend)
+
+        self.assertEqual(backend.close.call_count, 1)
+        self.assertEqual(backend.open.call_count, 2)
+        self.assertEqual(backend.connection.noop.call_count, 0)
+
+        # Old connection is open, we will not open a new connection because max connection lifetime is None.
+        with mock.patch.object(backend, "opened_at", timezone_now(), create=True), self.settings(
+            EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES=None
+        ):
+            initialize_connection(backend)
+
+        self.assertEqual(backend.close.call_count, 1)
+        self.assertEqual(backend.open.call_count, 3)
```

Comment:
...however, we do call `open` because that's how we check if the connection is _already_ open.  This probably deserves a comment because you just above say "we will not open a new connection" but then assert that we ... called `open`.  Which appears to contradict the above comment, and thus deserves its own comment.

Reply:
Added the comment and fixed!